src/ts/components/subtitletogglebutton.ts

```diff
@@ -0,0 +1,63 @@
+import { ToggleButton, ToggleButtonConfig } from './togglebutton';
+import { i18n } from '../localization/i18n';
+import { PlayerAPI } from 'bitmovin-player';
+import { UIInstanceManager } from '../uimanager';
+import { StorageUtils } from '../storageutils';
+import { SubtitleSelectBox } from './subtitleselectbox';
+import { SettingsPanel } from './settingspanel';
+
+
+export class SubtitleToggleButton extends ToggleButton<ToggleButtonConfig> {
+    private settingsPanel: SettingsPanel;
+    private subtitleSelectBox: SubtitleSelectBox;
+    private player: PlayerAPI;
+
+    constructor(subtitleSettingsOpenButton: SettingsPanel, subtitleSelectBox: SubtitleSelectBox, config: ToggleButtonConfig = {}) {
+        super(config);
+
+        this.settingsPanel = subtitleSettingsOpenButton;
+        this.subtitleSelectBox = subtitleSelectBox;
+
+        const defaultConfig: ToggleButtonConfig = {
+            cssClass: 'ui-subtitletogglebutton',
+            onClass: 'subtitles-on',
+            offClass: 'subtitles-off',
+            text: i18n.getLocalizer('settings.subtitles'),
+            ariaLabel: i18n.getLocalizer('settings.subtitles'),
+        };
+
+        this.config = this.mergeConfig(config, defaultConfig as ToggleButtonConfig, this.config);
+    }
+
+    configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
+        super.configure(player, uimanager);
+        this.player = player;
+        this.hide();
+
+        player.on(player.exports.PlayerEvent.SubtitleAdded, this.checkSubtitleAvailability);
+        player.on(player.exports.PlayerEvent.SubtitleRemoved, this.checkSubtitleAvailability);
+
+        this.onClick.subscribe(() => {
+            const availableSubtitles = player.subtitles.list();
+            const subtitleID = StorageUtils.getItem('bmpiu-subtitlelanguage');
```

Comment:
Two things here:
- The subtitle ID will be different for different videos, I am not sure that it's a good idea to store the ID. Maybe rather store a language + selected state (like `{ language: 'en', active: true }`) and try to find a matching subtitle for the new video.
Something like 
```ts
// const stored = {language: 'en', active: true};
if (stored.active) {
    const toActivate = player.subtitles.list().find(subtitle => subtitle.lang === stored.language);
    if (toActivate) {
        this.on(toActivate)
    } else {
        this.settingsPanel.show();
    }
}
```
- Do not use hard-coded prefix for `bmpiu-subtitlelanguage` rather use `DummyComponent.instance().prefixCss('subtitlelanguage');`